docs/resources/servicestage_component_instance.md

```diff
@@ -0,0 +1,426 @@
+  + **cci**: Cloud Container Instance (CCI)
+  + **ecs**: Elastic Cloud Server (ECS).
+  + **as**: Auto Scaling (AS)
+
+  The Optional resources include:
+  + **rds**: Relational Database Service (RDS)
+  + **dcs**: Distributed Cache Service (DCS),
+  + **elb**: Elastic Load Balance (ELB)
+
+  For other resource types, please refer to the environment documentation.
+
+* `id` - (Required, String) Specifies the resource ID.
+  If the `type` is set to **ecs**, the value of this parameter must be **Default**.
+
+* `alias` - (Optional, String) Specifies the application alias, which is provided only in DCS scenario.
+  The valid values are: **distributed_session**, **distributed_cache** and **distributed_session, distributed_cache**.
+  Defaults to **distributed_session, distributed_cache**.
+
+* `parameters` - (Optional, String) Specifies the reference resource parameter.
+  + When `type` is set to **cce**, this parameter is mandatory, and need to specify the namespace of the cluster where
+  the component is to be deployed, such as **{"namespace": "default"}**.
+  + When `type` is set to **ecs**, this parameter is mandatory, and need to specify the hosts where the component is to
+  be deployed, such as **{"hosts":["04d9f887-9860-4029-91d1-7d3102903a69", "04d9f887-9860-4029-91d1-7d3102903a70"]}}**.
+
+<a name="servicestage_artifact"></a>
+The `artifact` block supports:
+
+* `name` - (Required, String) Specifies the component name.
+  But for **Docker container scenario**, this name is the container name.
+
+* `type` - (Required, String) Specifies the source type.
+  The valid values are **package** (VM-based deployment) and **image** (container-based deployment).
+
+* `storage` - (Required, String) Specifies the storage mode. The valid values are **swr** and **obs**.
+
+* `url` - (Required, String) Specifies the software package or image address.
+  For a component deployed on a VM, this parameter is the software package address.
+  For a component deployed based on a container, this parameter is the image address or component name:v${index}.
+  The latter indicates that the component source code or the image automatically built using the software package
+  will be used.
+
+* `auth_type` - (Optional, String) Specifies the authentication mode.
+  The valid values are **iam** and **none**. Defaults to **iam**.
+
+* `version` - (Optional, String) Specifies the version number.
+
+<a name="servicestage_configuration"></a>
+The `configuration` block supports:
+
+* `env_variable` - (Optional, List) Specifies the environment variables.
+  The [object](#servicestage_env_variables) structure is documented below.
+
+-> After the instance is created, the system will automatically add a series of environment variables to it.
+   The specific parameters are subject to the state of the instance resource deployment.
+  This parameter will be affected by these variables and will appear when `terraform plan` or `terraform apply`.
```

Comment:
do we have a list of system env added to the instance by default? if the list is fixed we can help to ignore it for users.

Reply:
done.